### PR TITLE
Changing `content-type` from `application/json; charset=UTF-8` to `application/json;charset=UTF-8` unnecessarily

### PR DESCRIPTION
**Pull request by thiagohirata**
It breaks the signature, for instance, calculated by aws-amplify when making IAM_AWS authenticated calls.

### CODE REVIEW

**Comment by Zirro** [comment]
Thanks! Can you write a Web Platform Test showing that this change is accurate and behaves like other browsers, as described in our [Contributing.md](https://github.com/jsdom/jsdom/blob/master/Contributing.md)?

**Comment by thiagohirata** [comment]
Hi, @Zirro! WPT test ready!

File: test/web-platform-tests/to-upstream/xhr/content-type-without-charset.htm [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>Charset included into the content-type</title>
+    <script src="/resources/testharness.js"></script>
+    <script src="/resources/testharnessreport.js"></script>
+  </head>
+  <body>
+    <script type="text/javascript">
```
Comment (Zirro) [inline test/web-platform-tests/to-upstream/xhr/content-type-without-charset.htm]:
```suggestion
    <script>
```

Comment (Zirro) [reply test/web-platform-tests/to-upstream/xhr/content-type-without-charset.htm]: The `type` attribute is not needed.

File: test/web-platform-tests/to-upstream/xhr/content-type-without-charset.htm [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+      xhr.onload = test.step_func_done(function() {
+        assert_equals(xhr.status, 200);
+        assert_equals(xhr.responseText, "application/json; charset=UTF-8");
+      });
+      xhr.send("{ \"x\":\"a\"}");
+    });
+    </script>
+  </body>
+</html>
```
Comment (Zirro) [inline test/web-platform-tests/to-upstream/xhr/content-type-without-charset.htm]:
```suggestion
</html>

```

Comment (Zirro) [reply test/web-platform-tests/to-upstream/xhr/content-type-without-charset.htm]: Add a newline at end of file.

**Pull request merged**